src/workers/squadMemberJoinedMailing.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  getContactIdByEmail,
+  addUserToContacts,
+  LIST_DRIP_CAMPAIGN,
+} from '../common';
+import { SourceMember, SquadSource, User } from '../entity';
+import { SourceMemberRoles } from '../roles';
+import { ChangeObject } from '../types';
+import { messageToJson, Worker } from './worker';
+
+interface Data {
+  sourceMember: ChangeObject<SourceMember>;
+}
+
+const worker: Worker = {
+  subscription: 'api.member-joined-source-mailing',
+  handler: async (message, con, logger) => {
+    const data: Data = messageToJson(message);
+
+    try {
+      const { sourceMember } = data;
+
+      const [user, source] = await Promise.all([
+        con.getRepository(User).findOneBy({ id: sourceMember.userId }),
+        con.getRepository(SquadSource).findOneBy({ id: sourceMember.sourceId }),
+      ]);
+
+      // currently we treat admins as owners
+      const isOwner = sourceMember.role === SourceMemberRoles.Admin;
```

Comment:
It [was agreed](https://dailydotdev.slack.com/archives/C02E2C3C13R/p1693562363131039) we treat admins as owners for now

Reply:
Alternative implementation would be to listen for specific user action `create_squad` on cdc.

Reply:
Ah i thought for retro filling we could use admins, but for other cases i would use the new squad callback yes.
Sorry bit outdated on what this ticket is for again, ill check in more details tomorrow

Reply:
Yeah after thinking about it, since we introduce a new subscription let's make it a new topic.
Something like squad created, where we can internally even set a user action for squad_created (might come in handy)
and trigger this new subscription for now.

Think it's a bit more bulletproof.
As for the once-off we probably can simply export and import directly in sendgrid.

Reply:
@rebelchris I see all our topics are source related, so maybe `api.v1.source-created` and then we do specific logic for `type=squad`? I am little conflicted on implications of having more different topics so would appreciate feedback on how we do it usually.

Reply:
Yeah well could be source created or squad created depends where we put the logic.
Think a cdc that evalutates sources and matches type is lowest on resource.

@idoshamun would you agree?
Also happy to sync on this after my lunch ante?

Reply:
Yeah we can sync and then I can implement in one go.